### PR TITLE
clippy-deny the `todo!()` macro

### PR DESCRIPTION
**Pull request by problame**
`todo!()` shouldn't slip into prod code

### CODE REVIEW

**Comment by koivunej** [comment]
Confirmed, successfully denied the todo!.

**Comment by bojanserafimov** [comment]
This is not production code. See docstring of `LayerDescriptor`. The struct is in this module only because it's also used in benchmarks. Is there a better place to put this code that would satisfy the linter?

Replacing the `todo` with `unimplemented` is not a good move, because these shouldn't be implemented. They should be removed. But no big deal either way, it's testing code.

**Review by skyzh — APPROVED** [review]

**Comment by problame** [comment]
@bojanserafimov IIUC you abandoned #4393 in favor of work https://github.com/neondatabase/neon/pull/4398:

> No need for this. After https://github.com/neondatabase/neon/pull/4398 we can delete this struct

Can you follow up on the deletion of the struct so the `todo!()` goes away?

**Comment by skyzh** [comment]
We still need https://github.com/neondatabase/neon/pull/4437 before LayerDescriptor can be fully removed. On current main, LayerMap still needs a layer object instead of the descriptor.

**Comment by skyzh** [comment]
The upstream PR is merged and I can work on this soon

File: trace/src/main.rs [hunk 1]
```diff
@@ -164,8 +164,8 @@ fn main() -> anyhow::Result<()> {
                 analyze_trace(reader);
             }
         }
-        Command::Draw => todo!(),
-        Command::Replay => todo!(),
+        Command::Draw => unimplemented!(),
+        Command::Replay => unimplemented!(),
```
Comment (koivunej) [inline trace/src/main.rs:168]: I am late to the party but I think this hunk demonstrates that we don't really need this at all.

File: .neon_clippy_args [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # * `-A unknown_lints` – do not warn about unknown lint suppressions
 #                        that people with newer toolchains might use
 # * `-D warnings`      - fail on any warnings (`cargo` returns non-zero exit status)
-export CLIPPY_COMMON_ARGS="--locked --workspace --all-targets -- -A unknown_lints -D warnings"
+export CLIPPY_COMMON_ARGS="--locked --workspace --all-targets -- -A unknown_lints -D warnings -D clippy::todo"
```
Comment (arpad-m) [inline .neon_clippy_args:4]: Given that the comment above explains the two existing args, maybe the new arg should be explained in the comment above as well?

**Comment by skyzh** [comment]
clicked rerun button

**Pull request merged**